api/core/storage/slack/slack_api_client.py

```diff
@@ -15,63 +20,115 @@ def _client(self):
             headers={"Authorization": f"Bearer {self.bot_token}", "Content-Type": "application/json; charset=utf-8"},
         )
 
-    async def create_channel(self, name: str) -> str:
-        """Create a slack channel and return a channel id"""
+    def _check_response(
+        self,
+        parsed: dict[str, Any],
+        operation_name: str,
+    ) -> None:
+        """Check if Slack API response has ok=True, log and possibly raise error if not"""
+        if not parsed.get("ok", False):
+            error_msg = f"Slack client failed to {operation_name}"
+            error = parsed.get("error", "Unknown error")
+
+            _logger.error(error_msg, extra={"error_msg": error})
+            raise InternalError(error_msg, extra={"error": error})
+
+    async def post(
+        self,
+        endpoint: str,
+        json_data: dict[str, Any],
+        operation_name: str,
+    ) -> dict[str, Any]:
+        """Make a POST request to Slack API and check response"""
         async with self._client() as client:
-            response = await client.post(
-                "/conversations.create",
-                json={"name": name, "is_private": False},
+            response = await client.post(endpoint, json=json_data)
+            parsed = response.json()
+
+            self._check_response(
+                parsed,
+                operation_name,
             )
+
             response.raise_for_status()
+            return parsed
+
+    async def get(
+        self,
+        endpoint: str,
+        params: dict[str, str],
+        operation_name: str,
+    ) -> dict[str, Any]:
+        """Make a GET request to Slack API and check response"""
+        async with self._client() as client:
+            response = await client.get(endpoint, params=params)
             parsed = response.json()
```

Comment:
you can move that to the check response as well ?

Reply:
yes! https://github.com/WorkflowAI/WorkflowAI/pull/152/commits/ded9eb81f570efb64f3ac383117d9afe6a0cd75c